### DISCUSSION TITLE
🖥️ Self-hosted PWA

### DISCUSSION THREAD

**Discussion by 4gray** [4gray/iptvnator#211] [General]
### Hey folks,

I'm happy to announce that it is now possible to deploy the PWA version of IPTVnator on your own machine. This way you can install the player on e.g. Raspberry Pi, NAS or wherever you like. The docker images and docker-compose file have just been prepared for this case.  

Here you can find more details about the deployment: https://github.com/4gray/iptvnator/tree/electron/docker 

The backend part of the player, which is currently only responsible for parsing the playlists, is available in a separate repository, see  [iptvnator-backend](https://github.com/4gray/iptvnator-backend).

![docker-compose-iptvnator-self-hosted](https://user-images.githubusercontent.com/1503032/211679407-23810156-ad5a-4fbf-a30d-9f13ab489d8f.png)

As always, I look forward to your feedback 👋 

<hr /><em>This discussion was created from the release <a href='https://github.com/4gray/iptvnator/releases/tag/self-hosted'>🖥️ Self-hosted PWA</a>.</em>

**Comment by madejackson** [comment]
Are you serious!!! Finally someone who did it. Was looking for a webbased TV-Player with Catchup-Support like this for ages. Thank you so much. Will try it asap.

**Reply by madejackson** [reply]
Just tried on unraid: Docker Images are arm64 only. Could you also make them available for amd64? Thanks in advance :)

**Reply by berny6969** [reply]
only for arm64?

how to run on docker on amd64?

thanks!!!

**Reply by 4gray** [reply]
Thanks for the feedback, amd64 images were added 🎉

**Comment by StVol33** [comment]
Привет!
Не понятно как использовать и куда вставлять адрес EPG, например, такого формата: http://epg.it999.ru/edem.xml.gz

**Reply by 4gray** [reply]
Привет. В PWA версии пока что нет поддержки EPG, но планирую добавить в ближайшее время.

**Comment by aleksandrsma** [comment]
Hello,
why the playlist is not saving it is only cached.
When I try to log in with different device there is no playlist.
What may be the problem ?

**Reply by 4gray** [reply]
Hi, the playlists are stored client-side, on each device separately, other types of storage and synchronisation between devices is not supported yet.

**Comment by ErDongChen1973** [comment]
It can not play on webpage when i install it based on docker. What should i do next?
